File: src/main/java/com/widehouse/dnd5e/character/Ability.java
package com.widehouse.dnd5e.character;

import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public class Ability {
    Stat strength;
    Stat dexterity;
    Stat constitution;
    Stat intelligence;
    Stat wisdom;
    Stat charisma;

    /**
     * constructor.
     */
    public Ability(int str, int dex, int con, int intel, int wis, int cha) {
        this.strength = Stat.of(str);
        this.dexterity = Stat.of(dex);
        this.constitution = Stat.of(con);
        this.intelligence = Stat.of(intel);
        this.wisdom = Stat.of(wis);
        this.charisma = Stat.of(cha);
    }
}

File: src/main/kotlin/com/widehouse/dnd/character/ability/Ability.kt
package com.widehouse.dnd.character.ability

sealed class Ability(open val score: Int) {
    val modifier: Int
        get() = (score - 10) / 2
}

class Strength(override val score: Int) : Ability(score)
class Dexterity(override val score: Int) : Ability(score)
class Constitution(override val score: Int) : Ability(score)
class Intelligence(override val score: Int) : Ability(score)
class Wisdom(override val score: Int) : Ability(score)
class Charisma(override val score: Int) : Ability(score)
